qutebrowser/browser/webengine/webview.py

```diff
@@ -261,3 +264,30 @@ def acceptNavigationRequest(self,
                                                  is_main_frame=is_main_frame)
         self.navigation_request.emit(navigation)
         return navigation.accepted
+
+    def chooseFiles(self, mode, old_files, accepted_mimetypes):
+        """Override chooseFiles to (optionally) invoke custom file uploader."""
+        handler = config.val.fileselect.handler
+        if handler == "default":
+            return super().chooseFiles(mode, old_files, accepted_mimetypes)
+
+        with tempfile.NamedTemporaryFile(delete=False) as tmpfile:
+            filename = tmpfile.name
+
+        if mode == QWebEnginePage.FileSelectOpen:
+            command = config.val.fileselect.single_file.command
+        else:
+            command = config.val.fileselect.multiple_files.command
+
+        proc = guiprocess.GUIProcess(what='choose-file')
+        proc.start(command[0],
+                   [arg.replace('{}', filename) for arg in command[1:]])
+
+        loop = QEventLoop()
+        proc.finished.connect(lambda code, status: loop.exit())
+        loop.exec_()
+
+        with open(filename, 'r', encoding='utf8') as tmpfile:
```

Comment:
The encoding should probably be `sys.getfilesystemencoding()` since it's a filename?

Reply:
I will take a look at what happens when the encoding is not utf-8.

Reply:
@user202729 check for instance `cp1251` encoding. It is not the same as `utf-8`.